lib/source-map.js

```diff
@@ -369,8 +386,10 @@ SourceMap.prototype._scanMappings = function(srcMap, sourcesOffset, namesOffset,
         inputMappings = inputMappings.slice(match[0].length);
       }
     }
-
   }
+
+  // this.decoder.resetColumn();
+  this.encoder.resetColumn();
```

Comment:
its a tad tricky, once we started detecting failures this was required to fix it. I will add a comment though.